docs/programming-oak.md

```diff
@@ -537,12 +537,14 @@ function:
 ```Rust
 #[no_mangle]
 pub extern "C" fn frontend_oak_main(handle: u64) -> i32 {
-    match std::panic::catch_unwind(|| main(handle)) {
-        Ok(rc) => rc,
-        Err(_) => OakStatus::ERR_INTERNAL.value(),
-    }
+    std::panic::catch_unwind(|| main(handle))
+        .unwrap_or(Err(oak::OakStatus::ERR_INTERNAL))
+        .err()
+        .unwrap_or(oak::OakStatus::OK)
+        .value()
 }
-pub fn main(handle: u64) -> i32 {
+
+pub fn main(handle: u64) -> Result<(), oak::OakStatus> {
```

Comment:
There's some text further up (around l.507) that needs updating for `NodeMain` having a different signature.

Also, if we stick to having a different signature for inner main functions, we could do with some text to explain how/why.